src/app_service/service/stickers/service.nim

```diff
@@ -0,0 +1,383 @@
+      let balanceRpcResponse = self.getBalance(network.chainId, addressObj)
+
+      var balance = 0
+      if $balanceRpcResponse.result != "0x":
+        balance = parseHexInt(balanceRpcResponse.result.getStr)
+
+      var tokenIds: seq[int] = @[]
+
+      for it in toSeq[0..<balance]:
+        let response = self.tokenOfOwnerByIndex(network.chainId, addressObj, it.u256)
+        var tokenId = 0
+        if $response.result != "0x":
+          tokenId = parseHexInt(response.result.getStr)
+        tokenIds.add(tokenId)
+
+      var purchasedPackIds: seq[int] = @[]
+      for tokenId in tokenIds:
+        let response = self.getPackIdFromTokenId(network.chainId, tokenId.u256)
+        var packId = 0
+        if $response.result != "0x":
+          packId = parseHexInt(response.result.getStr)
+        purchasedPackIds.add(packId) 
+
+      self.purchasedStickerPacks = self.purchasedStickerPacks.concat(purchasedPackIds)
+      self.purchasedStickerPacks = self.purchasedStickerPacks.deduplicate()
+      result = self.purchasedStickerPacks
+    except RpcException:
+      error "Error getting purchased sticker packs", message = getCurrentExceptionMsg()
+      result = @[]
+
+  proc setAvailableStickerPacks*(self: Service, availableStickersJSON: string) {.slot.} =
+    let
+      accounts = self.walletAccountService.getWalletAccounts() # TODO: make generic
+      installedStickerPacks = self.getInstalledStickerPacks()
+    var
+      purchasedStickerPacks: seq[int]
+    for account in accounts:
+      purchasedStickerPacks = self.getPurchasedStickerPacks(account.address)
+    let availableStickers = JSON.decode($availableStickersJSON, seq[StickerPackDto])
+
+    let pendingTransactions = self.transactionService.getPendingTransactions()
+    var pendingStickerPacks = initHashSet[int]()
+    if (pendingTransactions != ""):
+      for trx in pendingTransactions.parseJson{"result"}.getElems():
+        if trx["type"].getStr == $PendingTransactionType.BuyStickerPack:
+          pendingStickerPacks.incl(trx["additionalData"].getStr.parseInt)
+
+    for stickerPack in availableStickers:
+      let isInstalled = installedStickerPacks.hasKey(stickerPack.id)
+      let isBought = purchasedStickerPacks.contains(stickerPack.id)
+      let isPending = pendingStickerPacks.contains(stickerPack.id) and not isBought
+      self.availableStickerPacks[stickerPack.id] = stickerPack
+      self.events.emit(SIGNAL_STICKER_PACK_LOADED, StickerPackLoadedArgs(
+        stickerPack: stickerPack,
+        isInstalled: isInstalled,
+        isBought: isBought,
+        isPending: isPending
+      )) 
+    self.events.emit(SIGNAL_ALL_STICKER_PACKS_LOADED, Args())
+
+  proc obtainAvailableStickerPacks*(self: Service) =
+    let networkType = self.settingsService.getCurrentNetwork().toNetworkType()
+    let network = self.networkService.getNetwork(networkType)
+    let contract = self.eth_service.findContract(network.chainId, "stickers")
```

Comment:
I think this might fail if there's no sticker pack contract available in the current network